main/src/ca/uwaterloo/flix/api/lsp/provider/completion/DeltaContext.scala

```diff
@@ -22,9 +22,31 @@ import ca.uwaterloo.flix.language.ast.Symbol
   */
 case class DeltaContext(deltas: List[Delta]) {
 
+  /**
+    * Merges this.List[Delta] with other.List[Delta].
+    * Removes duplicates that has an older timestamp.
+    *
+    * @param other the other DeltaContext.
+    * @return      a DeltaContext with the of changes as List[Delta],
+    *              where the most recent is the first elem of the list.
+    */
+  def mergeDeltas(other: DeltaContext): DeltaContext = {
+    val newDeltas = {
+      if (other.deltas.isEmpty) {
+        this.deltas
+      } else {
+        (this.deltas ++ other.deltas).reverse.distinctBy {
+          case Delta.ModifiedDef(sym, _) => sym
+          case Delta.AddEnum(sym, _) => sym
+          case Delta.AddField(field, _) => field
+        }.sortWith(_.timestamp > _.timestamp) // Sorts the list, to make sure the newest timestamp is the first elem.
+      }
+    }
+    DeltaContext(newDeltas)
+  }
```

Comment:
This gets the job done, but hopefully this can be optimized, however I'm not sure how...

Had to sort the list, because sometimes the list appeared to be randomized, and not in any order in terms of timestamps.

Need some feedback.

Reply:
It is a good question.

I would start by adding a helper function:

```
def filterOutdated(deltas: List[...]): List[...]
```

Reply:
The question is how to smartly remove events that are subsumed by newer events.

Reply:
Perhaps you could start by limiting the list to 1000 items and to the latest 5min